build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/DockerPull.java

```diff
@@ -0,0 +1,83 @@
+import org.jfrog.build.extractor.docker.DockerAgentUtils;
+import org.jfrog.build.extractor.packageManager.PackageManagerExtractor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.jfrog.build.extractor.packageManager.PackageManagerUtils.createArtifactoryClientConfiguration;
+
+public class DockerPull extends PackageManagerExtractor {
+    private Map<String, String> env;
+    private String username;
+    private String password;
+    private List<Module> modulesList = new ArrayList<>();
+    ArtifactoryDependenciesClientBuilder clientBuilder;
+    ArtifactoryClientConfiguration.BuildInfoHandler buildInfo;
+    String imageTag;
+    String host;
+    Log logger;
+
+    /**
+     * Docker push.
+     *
+     * @param clientBuilder        - Build Info client builder.
+     * @param imageTag             - Docker image to pull.
+     * @param host                 - Docker daemon ip.
+     * @param username             - The username to the docker registry.
+     * @param password             - The password to the docker registry.
+     * @param logger               - log output.
+     * @param env                  - Environment variables to use during docker push execution.
+     */
+    public DockerPull(ArtifactoryDependenciesClientBuilder clientBuilder, String imageTag, String host, String username, String password, Log logger, Map<String, String> env) {
+        this.clientBuilder = clientBuilder;
+        this.logger = logger;
+        this.imageTag = imageTag;
+        this.host = host;
+        this.username = username;
+        this.password = password;
+        this.env = env;
+    }
+
+    @Override
+    public Build execute() {
+        String message = "Pulling image: " + imageTag;
+        if (StringUtils.isNotEmpty(host)) {
+            message += " using docker daemon host: " + host;
+        }
+        logger.info(message);
+        DockerAgentUtils.pullImage(imageTag, username, password, host, env);
+        logger.info("Successfully pulled docker image: " + imageTag);
+        return new Build();
+    }
+
+    /**
+     * Allow running docker pull using a new Java process.
+     */
+    public static void main(String[] ignored) {
+        try {
+            ArtifactoryClientConfiguration clientConfiguration = createArtifactoryClientConfiguration();
+            ArtifactoryDependenciesClientBuilder clientBuilder = new ArtifactoryDependenciesClientBuilder().setClientConfiguration(clientConfiguration, clientConfiguration.publisher);
+            ArtifactoryClientConfiguration.DockerHandler dockerHandler = clientConfiguration.dockerHandler;
+            DockerPull dockerPush = new DockerPull(clientBuilder,
+                    dockerHandler.getImageTag(),
+                    dockerHandler.getHost(),
+                    clientConfiguration.publisher.getUsername(),
+                    clientConfiguration.publisher.getPassword(),
```

Comment:
We should use **clientConfiguration.resolver** and not **clientConfiguration.publisher** in the above two lines.
The same goes for the above line 68.